tests/DpTreeList.spec.js

```diff
@@ -0,0 +1,232 @@
+      draggable: true,
+      handleChange: jest.fn(),
+      handleDrag: jest.fn(),
+      checkBranch: mockCheckBranch,
+      level: 0,
+      node: {
+        "id": "1",
+        "type": "elements",
+        "attributes": {},
+        "children": [{
+          "id": "2",
+          "type": "elements",
+          "attributes": {},
+          "children": [{
+              "id": "3",
+              "type": "elements",
+              "attributes": {},
+              "children": [
+                {
+                  "attributes": {},
+                  "children": [],
+                  "id": "4",
+                  "type": "singleDocument"
+                },
+                {
+                  "attributes": {},
+                  "children": [],
+                  "id": "5",
+                  "type": "singleDocument"
+                }
+              ]
+            },
+            {
+              "id": "6",
+              "type": "elements",
+              "attributes": {},
+              "children": [{
+                  "attributes": {},
+                  "children": [],
+                  "id": "7",
+                  "type": "singleDocument"
+                }
+              ]
+            }
+          ]
+        }]
+      },
+      nodeId: '1',
+      onMove: jest.fn(),
+      options: {},
+      parentId: '',
+      parentSelected: false
+    }
+
+    mocks = {
+      Translator: {
+        trans: jest.fn(key => key)
+      }
+    }
+
+    stubs = {
+      'dp-tree-list-toggle': DpTreeListToggle
+    }
+  })
+
+  it('renders component correctly', () => {
+    const wrapper = shallowMount(DpTreeListNode, {
+      propsData,
+      mocks
+    })
+
+    expect(wrapper.exists()).toBe(true)
+  })
+
+  it('change the value of "isExpanded" to its opposite when toggling and change the icon class', async () => {
+    const wrapper = shallowMount(DpTreeListNode, {
+      propsData,
+      mocks,
+      stubs
+    })
+    const button = wrapper.findComponent(DpTreeListToggle)
+
+    expect(wrapper.vm.isExpanded).toBe(false)
+
+    await button.trigger('click')
+    expect(wrapper.vm.isExpanded).toBe(true)
+    expect(wrapper.vm.iconClassFolder).toBe('fa fa-folder-open')
+
+    await button.trigger('click')
+    expect(wrapper.vm.isExpanded).toBe(false)
+    expect(wrapper.vm.iconClassFolder).toBe('fa fa-folder')
+  })
+
+  it('add specific node properties (nodeId, nodeIsSelected, nodeType) to the node when invoking the setSelectionState method', () => {
+    const wrapper = shallowMount(DpTreeListNode, {
+      propsData,
+      mocks
+    })
+    const setNodeAndChildrenSelection = jest.spyOn(wrapper.vm, 'setNodeAndChildrenSelection')
+
+    wrapper.vm.setSelectionState(true)
+
+    expect(setNodeAndChildrenSelection).toHaveBeenCalled()
+
+    expect(wrapper.vm.node.nodeId).toBe(wrapper.vm.nodeId)
+    expect(wrapper.vm.node.nodeIsSelected).toBe(true)
+    expect(wrapper.vm.node.nodeType).toBe(wrapper.vm.isBranch ? 'branch' : 'leaf')
```

Comment:
Let's discuss this later in the fe unit test meeting. I think so too, that this is testing the implementation rather than the expected result which a user would have.